resources/xsd/fastRTPS_profiles.xsd

```diff
@@ -309,867 +1140,432 @@
-        <xs:all minOccurs="0">
-            <xs:element name="kind" type="historyQosKindType" minOccurs="0"/>
-            <xs:element name="depth" type="int32Type" minOccurs="0"/>
-        </xs:all>
-    </xs:complexType>
 
-    <xs:simpleType name="topicKindType">
-        <xs:restriction base="xs:string">
-            <xs:enumeration value="NO_KEY"/>
-            <xs:enumeration value="WITH_KEY"/>
-        </xs:restriction>
-    </xs:simpleType>
 
-    <xs:complexType name="topicAttributesType">
-        <xs:all minOccurs="0">
-            <xs:element name="kind" type="topicKindType" minOccurs="0"/>
-            <xs:element name="name" type="stringType" minOccurs="0"/>
-            <xs:element name="dataType" type="stringType" minOccurs="0"/>
-            <xs:element name="historyQos" type="historyQosPolicyType" minOccurs="0"/>
-            <xs:element name="resourceLimitsQos" type="resourceLimitsQosPolicyType" minOccurs="0"/>
-        </xs:all>
-    </xs:complexType>
 
-   <xs:simpleType name="durabilityQosKindType">
-        <xs:restriction base="xs:string">
-            <xs:enumeration value="VOLATILE"/>
-            <xs:enumeration value="TRANSIENT_LOCAL"/>
-            <xs:enumeration value="TRANSIENT"/>
-            <xs:enumeration value="PERSISTENT"/>
-        </xs:restriction>
-    </xs:simpleType>
 
-    <xs:complexType name="durabilityQosPolicyType">
+    <!--| QoS POLICY TYPES DEFINITION |-->
+    <!--QoS Data Sharing:
+        ├ kind         [string] ("AUTOMATIC", "ON", "OFF"),
+        ├ shared_dir   [string],
+        ├ domain_ids   [0~*],
+        |   └ domainID [uint32] (0~232)
+        └ max_domains  [uint32] -->
+    <xs:complexType name="dataSharingQosPolicyType">
         <xs:all>
-            <xs:element name="kind" type="durabilityQosKindType"/>
-        </xs:all>
-    </xs:complexType>
-
-    <xs:complexType name="durabilityServiceQosPolicyType">
-        <xs:all minOccurs="0">
-            <xs:element name="service_cleanup_delay" type="durationType" minOccurs="0"/>
-            <xs:element name="history_kind" type="historyQosKindType" minOccurs="0"/>
-            <xs:element name="history_depth" type="uint32Type" minOccurs="0"/>
-            <xs:element name="max_samples" type="uint32Type" minOccurs="0"/>
-            <xs:element name="max_instances" type="uint32Type" minOccurs="0"/>
-            <xs:element name="max_samples_per_instance" type="uint32Type" minOccurs="0"/>
+            <xs:element name="kind" minOccurs="1" maxOccurs="1">
+                <xs:simpleType>
+                    <xs:restriction base="xs:string">
+                        <xs:enumeration value="AUTOMATIC"/>
+                        <xs:enumeration value="ON"/>
+                        <xs:enumeration value="OFF"/>
+                    </xs:restriction>
+                </xs:simpleType>
+            </xs:element>
+            <xs:element name="shared_dir" type="string" minOccurs="0" maxOccurs="1"/>
```

Comment:
I have tried erasing the value inside a "shared_dir" tag, and the validator says it is valid.